pkg/prometheus/promcfg_test.go

```diff
@@ -2003,19 +2081,13 @@ func TestAdditionalScrapeConfigs(t *testing.T) {
 		cg := mustNewConfigGenerator(t, p)
 
 		cfg, err := cg.Generate(
-			&monitoringv1.Prometheus{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "test",
-					Namespace: "default",
-				},
-				Spec: monitoringv1.PrometheusSpec{
-					CommonPrometheusFields: monitoringv1.CommonPrometheusFields{
-						Shards:         shards,
-						ScrapeInterval: "30s",
-					},
-					EvaluationInterval: "30s",
-				},
-			},
+			"30s",
```

Comment:
nit: maybe create the Prometheus object before like for the other tests? I find it easier to read/understand what we're testing.

Reply:
I just noticed that we have a variable `p` of type Prometheus just some lines above. I'm not sure why a new one was being created 😅